src/axolotl/logging_config.py

```diff
@@ -23,6 +23,7 @@ class ColorfulFormatter(Formatter):
     }
 
     def format(self, record):
+        record.rank = int(os.getenv("LOCAL_RANK", "0"))
```

Comment:
Seems good. I was a bit confused sometimes which rank the output comes from.

When this is multi-node, this can have duplicate values, therefore, we might need global_rank when we move into multi-node.